### DISCUSSION TITLE
Why does `push_to_git_remote` have seemingly needless call to `pwd`?

### DISCUSSION THREAD

**Discussion by kohenkatz** [fastlane/fastlane#18231] [General] [answered]
I have been so happy with fastlane for iOS apps that last year I decided to use it for Android apps too. So far that has worked very well, and it even works on Windows!

Today I wrote a new lane that guides the user through our release process, which is based on [gitflow-avh](https://github.com/petervanderdoes/gitflow-avh). At the end, it runs the following fastlane actions:

```
push_git_tags(tag: release_tag)
push_to_git_remote(local_branch: 'develop')
push_to_git_remote(local_branch: 'master')
```

On Windows, the `push_to_git_remote` calls fail with this errror:

```
[18:29:12]: --------------------------------
[18:29:12]: --- Step: push_to_git_remote ---
[18:29:12]: --------------------------------
[18:29:12]: $ pwd
+-----------------------------+-----------------+
|                 Lane Context                  |
+-----------------------------+-----------------+
| DEFAULT_PLATFORM            | android         |
| PLATFORM_NAME               | android         |
| LANE_NAME                   | android release |
| GIT_REPO_WAS_CLEAN_ON_START | true            |
+-----------------------------+-----------------+
[18:29:12]: No such file or directory - pwd
```

I was understandably baffled as to where this call to `pwd` comes from, since I can't think of any reason for it to be there. It turns out that a call to `Actions.sh('pwd')` has been present in the code of this action [since its creation 6 years ago](https://github.com/fastlane/fastlane/commit/9008df0095fb4a6046c20e81aa0a33165969ab46).

It does not seem like there is any reason this call needs to be here, since its output is never used for anything. It seems like it was probably a debugging line that was never removed. In any case, it breaks a function that would otherwise work perfectly on Windows. (Compare to `push_git_tags`, which *does* work on Windows.)

I would submit a PR to simply remove that line, but I thought I would ask first if there's a reason it needs to be there. (If that's the case, I'll replace it with `Dir.pwd` instead.)

**Comment by ainame** [comment, marked as answer]
I don't really know the truth but it looks like it's been sitting down there just for a debugging purpose when implemented from the very early stage of fastlane. Let's clean it up 💪 
https://github.com/fastlane/fastlane/pull/111

**Reply by kohenkatz** [reply]
#18270
